### DISCUSSION TITLE
Do you plan to add some primitive like styled-system ?

### DISCUSSION THREAD

**Discussion by LoiKos** [stitchesjs/stitches#499] [Help] [answered]
Hi there! I just discover stitches a few days ago and it look really promising. 

I just wonder if you plan to add some primitive like [styled-system](https://styled-system.com/) do (e.g to build some utils components like `Box`, `Flex`, `Text` ) ?

**Comment by peduarte** [comment, marked as answer]
Hey! Thanks for checking it out.

Stitches itself won't ship with primitives. We'd like to keep the codebase/bundle size as small as possible, focusing solely on the styling solution. 

These primitives are highly dependant on the users config (`theme`, `media`, `utils`), so I think it's clearer if they live product-side.

Primitives like `Box` and `Flex` are relatively low-maintenance:
```jsx
const Box = styled('div')
const Flex = styled('div', {
  display: 'flex'
})
```

But keeping them product side also gives you more control over which properties/variants you want to bake in!

**Reply by LoiKos** [reply]
ok that make sense ! Thank for the quick answer :)
